packages/create-vite/index.js

```diff
@@ -237,6 +263,33 @@ async function init() {
   // determine template
   template = variant || framework || template
 
+  const pkgInfo = pkgFromUserAgent(process.env.npm_config_user_agent)
+  const pkgManager = pkgInfo ? pkgInfo.name : 'npm'
+
+  if (template.startsWith('custom-')) {
+    const getCustomCommand = (name) => {
+      for (const f of FRAMEWORKS) {
+        for (const v of f.variants || []) {
+          if (v.name === name) {
+            return v.customCommand
+          }
+        }
+      }
+    }
+    const customCommand = getCustomCommand(template)
+    const fullCustomCommand = customCommand
+      .replace('TARGET_DIR', targetDir)
+      .replace(/^npm /, `${pkgManager} `)
+      // Only yarn doesn't support `@version` in the `create` command
+      .replace('@latest', () => (pkgManager === 'yarn' ? '' : '@latest'))
+
+    const [command, ...args] = fullCustomCommand.split(' ')
+    const { status } = spawnSync(command, args, {
+      stdio: 'inherit'
+    })
+    process.exit(status ?? 0)
```

Comment:
I tested on Windows with powershell/cmd and the script finished without showing the `create-vue`.
The `error` of `spawnSync` was:
```
Error: spawnSync npm ENOENT
    at Object.spawnSync (node:internal/child_process:1111:20)
    at spawnSync (node:child_process:814:24)
    at init (file:///D:/documents/GitHub/vite/packages/create-vite/index.js:287:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {        
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawnSync npm',
  path: 'npm',
  spawnargs: [ 'create', 'vue@latest', 'vite-project' ]
}
```

Maybe we need to append `.cmd` to `pkgManager` variable when running on Windows.
Or use [`cross-spawn`](https://github.com/moxystudio/node-cross-spawn), which respects PATHEXT.

Reply:
Thanks! I fixed it by using `cross-spawn`. It's a very small package so I think it wouldn't hurt the overall user experience: https://bundlephobia.com/package/cross-spawn@7.0.3

Reply:
We could make it even smaller by #9034. 🙂